### ISSUE TITLE
nn.functional should maintain API parity with nn

### ISSUE THREAD

**Issue by narfanar** [pytorch/pytorch#30413]
## 🚀 Feature
`nn.functional` should provide a functional alternative to every stateful component in `nn` that supports it.

Some, like `Tanh() -> tanh` were deprecated in `nn.functional` because they're now the same as those in `torch`.

## Motivation

People usually import modules and submodules that are used often under short intuitive names.

One such name I've often encountered is `nn.functional as F`, in the same vein as importing `nn`.

The current plan to move some math functions permanently away from `nn.functional` somewhat awkwardly splits the API.

1. Every operation that has a `Module` has its name in `nn` (to my knowledge).
2. Almost every (all?) operation `Module` in `nn` that can be exposed cleanly in a functional manner has a name in `nn.functional`.

## Pitch

Move all 'mathy' functions with class counterparts from `torch` to `nn.functional` and expose aliases to the basic/fundamental ones in `torch`.

Here we get into ugly territory; I don't know how exactly I'd argue for some methods currently under `torch`, but `tanh`, `sigmoid` and co definitely belong in `functional`.

## Alternatives

Flip the implementation/re-exporting tale with the impls in `torch` and the re-exports in `nn.functional`.

## Additional context

N/A
